src/main/php/text/encode/HHVMBase64OutputStream.class.php

```diff
@@ -0,0 +1,115 @@
+<?php namespace text\encode;
+
+use io\streams\OutputStream;
+use io\streams\Streams;
+use lang\Type;
+
+/**
+ * OuputStream that encodes data to base64 encoding
+ *
+ * @see      rfc://2045 section 6.8 
+ * @test     xp://net.xp_framework.unittest.text.encode.Base64OutputStreamTest
+ */
+class HHVMBase64OutputStream implements OutputStream {
+  const TABLE = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/';
+  const PAD = '=';
+
+  private $out, $remain;
+  private $pos= 0;
+  
+  /**
+   * Constructor
+   *
+   * @param   io.streams.OutputStream out
+   * @param   int lineLength limit maximum line length
+   */
+  public function __construct(OutputStream $out, $lineLength= 0) {
+    $this->out= $out;
+    $this->remain= '';
+    if ($lineLength) {
+      $pos= 0;
+      $this->write= function($bytes) use(&$pos, $lineLength) {
+        $l= strlen($bytes);
+        $o= 0;
+        while ($l - $o > $lineLength - $pos) {
+          $this->out->write(substr($bytes, $o, $lineLength - $pos)."\n");
+          $pos= 0;
+          $o+= $lineLength;
+        }
+        $this->out->write(substr($bytes, $o));
+        $pos= $l - $o;
+      };
+    } else {
+      $this->write= Type::forName('function(var): var')->cast([$this->out, 'write']);
```

Comment:
This is essentially `Closure::fromCallable()`, but this PHP 7.1 feature doesn't exist in HHVM yet, see https://github.com/facebook/hhvm/issues/7874